--- httpclient.rb~	2007-09-12 19:21:01.000000000 -0400
+++ httpclient.rb	2007-10-19 14:59:06.000000000 -0400
@@ -1823,7 +1823,7 @@
   def default_redirect_uri_callback(uri, res)
     newuri = URI.parse(res.header['location'][0])
     unless newuri.is_a?(URI::HTTP)
-      newuri = uri + newuri
+      newuri = URI.join(uri, newuri)
       STDERR.puts(
         "could be a relative URI in location header which is not recommended")
       STDERR.puts(
